### PR TITLE
Fix up busybox detection for relative symlinks

### PR DESCRIPTION
**Pull request by jonmeow**
Handling for relative symlinks is new (comment talks about relative symlinks). Relocated to add tests though, to do extra checking of logic.

### CODE REVIEW

**Review by danakj — APPROVED** [review]

**Review by josh11b — APPROVED** [review]

File: toolchain/install/busybox_info.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    std::error_code ec;
+    auto symlink_target = std::filesystem::read_symlink(info.bin_path, ec);
+    if (ec) {
+      return ErrorBuilder()
+             << "expected carbon-busybox symlink at `" << info.bin_path << "`";
+    }
+    info.mode = filename;
+    // Do a path join, to handle relative symlinks.
+    info.bin_path = info.bin_path.parent_path() / symlink_target;
```
Comment (chandlerc) [inline toolchain/install/busybox_info.h:41]: This *only* handles relative symlinks. That's certainly what we need at the moment, but do you want to support both? Worth documenting only relative?

**Review by chandlerc — COMMENTED** [review]
Sorry for the late comment, but also not critical...

File: toolchain/install/busybox_info.h (hunk 1 above)
Comment (jonmeow) [reply toolchain/install/busybox_info.h:41]: What do you mean "only"? My understanding is this works for absolute, copying from https://en.cppreference.com/w/cpp/filesystem/path/append:

```cpp
path("foo") / "/bar"; // the result is "/bar" (replaces)
```

Comment (jonmeow) [reply toolchain/install/busybox_info.h:41]: Should have an explicit test though, so sent https://github.com/carbon-language/carbon-lang/pull/4528

**Pull request merged**